src/TuDa.CIMS.Api/Repositories/WorkingGroupRepository.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TuDa.CIMS.Api.Database;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+
+namespace TuDa.CIMS.Api.Repositories;
+
+public class WorkingGroupRepository : IWorkingGroupRepository
+{
+    private readonly CIMSDbContext _context;
+
+    public WorkingGroupRepository(CIMSDbContext context)
+    {
+        _context = context;
+    }
+
+    /// <summary>
+    /// Returns an existing Working Group with the specific id.
+    /// </summary>
+    /// <param name="id">the specific id of the Working Group</param>
+    /// <returns></returns>
+    public async Task<WorkingGroup?> GetOneAsync(Guid id)
+    {
+        return await _context.WorkingGroups.Where(i => i.Id == id).SingleOrDefaultAsync();
+    }
+
+    /// <summary>
+    /// Returns all existing Working Groups of the database.
+    /// </summary>
+    public async Task<IEnumerable<WorkingGroup>> GetAllAsync()
+    {
+        return await _context.WorkingGroups.Include(workingGroup => workingGroup.Professor)
+            .Include(workingGroup => workingGroup.Students).Include(workingGroup => workingGroup.Purchases)
+            .ToListAsync();
+    }
+
+
+    /// <summary>
+    /// Updates an existing Working Group with the specified id using the provided update model.
+    /// </summary>
+    /// <param name="id">the specific id of the Working Group</param>
+    /// <param name="updateModel">the model containing the updated values for the Working Group </param>
+    public async Task<ErrorOr<Updated>> UpdateAsync(Guid id, UpdateWorkingGroupDto updateModel)
+    {
+        var existingItem = await _context
+            .WorkingGroups.Include(workingGroup => workingGroup.Professor)
+            .Include(workingGroup => workingGroup.Students).Include(workingGroup => workingGroup.Purchases)
+            .SingleOrDefaultAsync(i => i.Id == id);
```

Comment:
🌱 For performance reason, you should move the where from the `SingleOrDefault` to an actual `Where` method and move it to the top, so the Include only is done on the one Item